main.go

```
@@ -72,6 +88,54 @@ func ToEther(o *big.Int) *big.Float {
 	return pul
 }
 
+func CallContractFunction(contractAddress string, abiString string, functionName string) (string, error) {
+	// Parse ABI
+	contractABI, err := abi.JSON(strings.NewReader(abiString))
+	if err != nil {
+		return "", fmt.Errorf("failed to parse ABI: %v", err)
+	}
+
+	// Pack the function call
+	data, err := contractABI.Pack(functionName)
+	if err != nil {
+		return "", fmt.Errorf("failed to pack function call: %v", err)
+	}
+
+	// Create call message
+	contractAddr := common.HexToAddress(contractAddress)
+	msg := ethereum.CallMsg{
+		To:   &contractAddr,
+		Data: data,
+	}
+
+	// Check if eth client is connected
+	if eth == nil {
+		return "", fmt.Errorf("failed to call contract: eth client not connected")
+	}
+
+	// Call the contract
+	result, err := eth.CallContract(context.TODO(), msg, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to call contract: %v", err)
+	}
+
+	// Unpack the result
+	outputs, err := contractABI.Unpack(functionName, result)
+	if err != nil {
+		return "", fmt.Errorf("failed to unpack result: %v", err)
+	}
+
+	// Convert result to string
+	if len(outputs) > 0 {
+		if bigInt, ok := outputs[0].(*big.Int); ok {
+			return bigInt.String(), nil
+		}
+		return fmt.Sprintf("%v", outputs[0]), nil
+	}
+
+	return "0", nil
```

Is there a valid case where `len(outputs)` is not greater than zero? Wouldn't it give false-positive result if we return `0`? Shouldn't we throw an error in such case?

---

Good call, we should throw an error: [ffc7762](https://github.com/mezo-org/balance-exporter/pull/1/commits/ffc776267edd112e9b2fe358d91ba7eaca966293)